src/k8s/pkg/utils/net.go

```diff
@@ -0,0 +1,30 @@
+package utils
+
+import (
+	"errors"
+	"net"
+	"os"
+	"syscall"
+	"time"
+)
+
+// IsLocalPortOpen checks if the given local port is already open or not.
+func IsLocalPortOpen(port string) (bool, error) {
+	if err := checkPort("localhost", port, 500*time.Millisecond); err == nil {
```

Comment:
Have you verified if this check is sufficient for the following conditions:

1. Port bound on all IPv4 addresses (0.0.0.0)
2. Port bound on all IPv6 addresses (::)
3. Port bound to the node IP address (e.g. 10.0.0.49)
4. Port bound to IPv4 localhost (localhost/127.0.0.1)
5. Port bound to IPv6 link-local (ff80::)

Reply:
Done